### ISSUE TITLE
Styleguide/best practices

### ISSUE THREAD

**Issue by mcupak** [ga4gh/TASC#6]
Some time ago, I started writing down "mundane" things we try to do consistently across several repositories, sort of a [styleguide/getting started guide](https://github.com/ga4gh-discovery/ga4gh-styleguide) together with a [template repo](https://github.com/mcupak/ga4gh-specification-template). Would this document or a subset of its items be of interest to this team? If so, what do we need to do to make it more official and further evolve it together?

**Comment by jb-adams** [comment]
@mcupak looks good to me. I think building out this info and making it available across work streams would likely be quite beneficial. A styleguide might fall under the realm of "Best Practices." Like standards, Best Practices can be proposed to steering committee (@rishidev please correct me if I'm wrong).

**Comment by jmarshall** [comment]
Some of this (e.g., REST, OpenAPI) is only applicable to the subset of GA4GH products that are network protocols.

Other items, notably licensing and versioning, are major items that would perhaps best be considered individually. There is a REWS sub-group on IP that would perhaps cover licensing. Versioning IIRC is something that it has been suggested that TASC could usefully consider and provide suggestions for.

I see TASC's role as aiding the harmonisation of aspects of GA4GH's various products that would otherwise prevent different products from being used together conveniently. To the extent that a style guide would be about how working groups and committees organised and did their work (rather than about the end products themselves), I would think that was outside TASC's remit.

**Comment by mcupak** [comment]
Thanks for the comments! I view this more as a list of common patterns rather than a prescriptive document you'd have to follow. Completely fair if this is out of scope, I'm certainly thinking of it more as an internal resource for developers within GA4GH rather than outside users consuming the products.

**Comment by rishidev** [comment]
I agree this is out of scope here, agreeing with points made by jmarshall above. There is some overlap with the non-styleguide section with the [Product Approval Process document Appendix B as well](https://w3id.org/ga4gh/product-approval).
